Neos.Flow/Classes/Mvc/Routing/Dto/UriConstraints.php

```diff
@@ -320,9 +320,13 @@ public function applyTo(UriInterface $baseUri, bool $forceAbsoluteUri): UriInter
             $uri = $uri->withPath(trim($baseUri->getPath(), '/') . '/' . ltrim($uri->getPath(), '/'));
         }
 
-        // Ensure the URL always starts with "/" if non-empty.
+        // Ensure the URL always starts with "/" if non-empty; and if path is empty, set it to "/" in order to
+        // enforce it *ALWAYS* starts with a /. The 2nd case is needed to e.g. link to the root node of a Neos
+        // site correctly.
         if (strlen($uri->getPath()) > 0 && $uri->getPath(){0} !== '/') {
```

Comment:
`If (strlen() === 0 || getPath(){0} != '/') {` maybe to avoid the elseif?

Reply:
changed.